backend/src/main/java/ch/puzzle/okr/service/validation/ObjectiveValidationService.java

```diff
@@ -36,4 +41,13 @@ public void validateOnUpdate(Long id, Objective model) {
         doesEntityExist(id);
         validate(model);
     }
+
+    @Override
+    public void validateOnDelete(Long id) {
+        throwExceptionWhenIdIsNull(id);
+        doesEntityExist(id);
+        if (checkInBusinessService.getCheckInAmountByObjectiveId(id) != 0) {
```

Comment:
this error should not be thrown, a delete on an objective should delete its corresponding key results and the corresponding check ins